### PR TITLE
Revamp index

### PR DESCRIPTION
**Pull request by dmp42**
Thanks to Mary and Stephen hard work, we have a solid basis for the documentation on top of which we can now iterate.

This first attempt at enhancing things is rather modest, and focused on a couple of key items:

 * shortening things: making the message easier to digest by removing redundant information, avoidable expansions, etc
 * clearer message: with more focus on what matters from a product POV

Let me know if you feel this is going in the right direction and if we want to go further.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
 page_keywords: registry, images, repository
 IGNORES-->
 
-# Docker Registry 2.0
+# Docker Registry
 
-Docker Registry stores and distributes images centrally. It's where you push images to and pull them from; Docker Registry gives team members the ability to share images and deploy them to testing, staging and production environments.
```
Comment (dmp42) [inline docs/index.md]: The team members and testing/staging/production thing feels irrelevant.
So does words like "centrally".

File: docs/index.md [hunk 2]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
-Docker provides a hosted registry as part of [Docker Hub](https://hub.docker.com). Docker Hub is a cloud service that securely manages your images. It features organization accounts, automated builds, and much, much more.
+Users looking for a ready-to-go solution are encouraged to head-over to the [Docker Hub](https://hub.docker.com), which does provide a free-to-use, hosted Registry, plus additional features (organization accounts, automated builds, and more).
 
-Docker Registry is the core technology behind the Docker Hub. You can run your own registry instance if you want to host your images privately. Docker Registry features:
-
- - **Pluggable storage drivers**: Images can be stored in Amazon S3, Microsoft Azure or the local filesystem.
- - **Webhook notifications**: When an image is pushed to your registry, webhooks can fire off to launch CI builds, send notifications to IRC, etc.
-
-To get started with your own Docker Registry, head over to the instructions on how to [deploy a registry](deploying.md).
+On the other hand, people interested in finer grained integration, or more control over the content they publish, should run and operate their own Registry.
```
Comment (dmp42) [inline docs/index.md]: Trying to clarify things with regard to Hub in two sentences.

1. what is hub and why you want it
2. why you would want more and why the registry is for you then

File: docs/index.md [hunk 3]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
+The Registry is a stateless, highly scalable server side component that stores and lets you distribute Docker images.
 
-Docker provides a hosted registry as part of [Docker Hub](https://hub.docker.com). Docker Hub is a cloud service that securely manages your images. It features organization accounts, automated builds, and much, much more.
+Users looking for a ready-to-go solution are encouraged to head-over to the [Docker Hub](https://hub.docker.com), which does provide a free-to-use, hosted Registry, plus additional features (organization accounts, automated builds, and more).
 
-Docker Registry is the core technology behind the Docker Hub. You can run your own registry instance if you want to host your images privately. Docker Registry features:
-
- - **Pluggable storage drivers**: Images can be stored in Amazon S3, Microsoft Azure or the local filesystem.
- - **Webhook notifications**: When an image is pushed to your registry, webhooks can fire off to launch CI builds, send notifications to IRC, etc.
```
Comment (dmp42) [inline docs/index.md]: This is already going into "understanding the registry" and feels like it belongs to below.

File: docs/index.md [hunk 4]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
-
-To get started with your own Docker Registry, head over to the instructions on how to [deploy a registry](deploying.md).
```
Comment (dmp42) [inline docs/index.md]: Duplicated (below).

File: docs/index.md [hunk 5]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
+On the other hand, people interested in finer grained integration, or more control over the content they publish, should run and operate their own Registry.
 
 ## Understanding the registry
 
-A registry is, at its heart, a collection of repositories. In turn, a repository
-is collection of images. Users interact with the registry by pushing images to
-or pulling images from the registry. The Docker Registry includes several
-optional features that you can configure according to your needs.
+A Registry is a storage and content delivery system, holding named Docker images, available in different tagged versions. For example, the image `distribution/registry`, with tags `2.0` and `latest`.
```
Comment (dmp42) [inline docs/index.md]: Trying to make things simpler than "repository" (ambiguous wrt to github vs. docker) and "collections".

File: docs/index.md [hunk 6]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
 
-![](images/registry.png)
```
Comment (dmp42) [inline docs/index.md]: This image does not make sense to me.

File: docs/index.md [hunk 7]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
+Users interact with a registry by using docker push and pull commands. For example, `docker pull myregistry.com/stevvooe/batman:voice`.
 
-The architecture supports a configurable storage backend. You can store images
-on  a file system or on a service such as Amazon S3 or Microsoft Azure. The
-default storage system is the local disk; this is suitable for development or
-some small deployments.
+Storage itself is delegated to drivers. The default storage driver is the local posix filesystem, which is suitable for development or small deployments. Additional cloud-based storage driver like S3 and Microsoft Azure are also supported. People looking into using other storage backends may do so by implementing their own driver following the [Storage API](storagedrivers.md).
```
Comment (dmp42) [inline docs/index.md]: This and below is just a bit more crisp, but keeps the same ideas/features.

File: docs/index.md [hunk 8]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
-sends webhook notifications in response to registry activity.  The registry also
-includes features for both logging and reporting as well. Reporting is useful
-for large installations that want to collect metrics. Currently, the feature
-supports both New Relic and Bugsnag.
+Finally, the Registry includes a robust [notification system](notifications.md), calling webhooks in response to activity, and both extensive logging and reporting. Reporting is mostly useful for large installations that want to collect metrics. Currently, New Relic and Bugsnag are supported.
 
 ## Getting help
 
-Docker Registry is an open source project and is under active development. If
-you need help, would like to contribute, or simply want to talk about the
```
Comment (dmp42) [inline docs/index.md]: 80 chars line breaks are non-sensical.

File: docs/index.md [hunk 9]
```diff
@@ -59,10 +36,15 @@ communication.
 
 For more information and resources, please visit the [Getting Help project page](https://docs.docker.com/project/get-help/).
 
-## Registry documentation
+## Documentation
+
+Basics:
```
Comment (dmp42) [inline docs/index.md]: In the interest of keeping the documentation volume at bay, we should really aim for a separation between use cases. Casual users need no more than deploy and configure.
Everything else is advanced use, and does not deserve a front row seat.

File: docs/index.md [hunk 10]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
-# Docker Registry 2.0
+# Docker Registry
 
-Docker Registry stores and distributes images centrally. It's where you push images to and pull them from; Docker Registry gives team members the ability to share images and deploy them to testing, staging and production environments.
+The Registry is a stateless, highly scalable server side component that stores and lets you distribute Docker images.
 
-Docker provides a hosted registry as part of [Docker Hub](https://hub.docker.com). Docker Hub is a cloud service that securely manages your images. It features organization accounts, automated builds, and much, much more.
+Users looking for a ready-to-go solution are encouraged to head-over to the [Docker Hub](https://hub.docker.com), which does provide a free-to-use, hosted Registry, plus additional features (organization accounts, automated builds, and more).
 
-Docker Registry is the core technology behind the Docker Hub. You can run your own registry instance if you want to host your images privately. Docker Registry features:
```
Comment (dmp42) [inline docs/index.md]: Confusing. Hub also supports private images.

**Comment by aluzzardi** [comment]
This looks much more clear and concise as a getting started point :+1: 

We are trying to do an overhaul of the Swarm docs as well. Just like the Registry, a lot has changed since the documentation was initially written and an update is overdue.

LGTM

**Comment by stevvooe** [comment]
LGTM!

**Comment by stevvooe** [comment]
@spf13 @moxiegirl PTAL

File: docs/index.md [hunk 11]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
 IGNORES-->
 
-# Docker Registry 2.0
+# Docker Registry
 
-Docker Registry stores and distributes images centrally. It's where you push images to and pull them from; Docker Registry gives team members the ability to share images and deploy them to testing, staging and production environments.
+The Registry is a stateless, highly scalable server side component that stores and lets you distribute Docker images.
 
-Docker provides a hosted registry as part of [Docker Hub](https://hub.docker.com). Docker Hub is a cloud service that securely manages your images. It features organization accounts, automated builds, and much, much more.
+Users looking for a ready-to-go solution are encouraged to head-over to the [Docker Hub](https://hub.docker.com), which does provide a free-to-use, hosted Registry, plus additional features (organization accounts, automated builds, and more).
```
Comment (NathanMcCauley) [inline docs/index.md]: s/which does provide/which provides/

File: docs/index.md [hunk 12]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
-on  a file system or on a service such as Amazon S3 or Microsoft Azure. The
-default storage system is the local disk; this is suitable for development or
-some small deployments.
+Storage itself is delegated to drivers. The default storage driver is the local posix filesystem, which is suitable for development or small deployments. Additional cloud-based storage driver like S3 and Microsoft Azure are also supported. People looking into using other storage backends may do so by implementing their own driver following the [Storage API](storagedrivers.md).
 
-Securing access to images is a concern for even the simplest deployment. The
-registry service supports transport layer security (TLS) natively. You must
-configure it in your instance to make use of it. You can also use a proxy server
-such as Nginx and basic authentication to extend the security of a deployment.  
+Since securing access to your hosted images is paramount, the Registry does natively support TLS. You can also enforce basic authentication through a proxy like Nginx.
```
Comment (NathanMcCauley) [inline docs/index.md]: remove 'does'

**Comment by NathanMcCauley** [comment]
LGTM after nits

**Comment by dmp42** [comment]
@NathanMcCauley nits picked.

**Comment by RichardScothern** [comment]
A small nit: 'registy' should have a capital 'r' in all places, since it's a proper name.  esp. in the second heading.

File: docs/index.md (hunk 7 above)
Comment (RichardScothern) [inline docs/index.md]: developing their own driver implementing the [Storage API] ?

**Comment by RichardScothern** [comment]
I like it, very succinct.  LGTM!

**Comment by dmp42** [comment]
@RichardScothern nits picked.

File: docs/index.md [hunk 13]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
 
-Docker Registry is the core technology behind the Docker Hub. You can run your own registry instance if you want to host your images privately. Docker Registry features:
+On the other hand, people interested in finer grained integration, or more control over the content they publish, should run and operate their own Registry.
 
- - **Pluggable storage drivers**: Images can be stored in Amazon S3, Microsoft Azure or the local filesystem.
- - **Webhook notifications**: When an image is pushed to your registry, webhooks can fire off to launch CI builds, send notifications to IRC, etc.
+## Understanding the Registry
 
-To get started with your own Docker Registry, head over to the instructions on how to [deploy a registry](deploying.md).
+A Registry is a storage and content delivery system, holding named Docker images, available in different tagged versions. For example, the image `distribution/registry`, with tags `2.0` and `latest`.
```
Comment (thaJeztah) [inline docs/index.md]: If "A Registry" here refers to a registry "in general", not the "Docker" registry, I don't think it should be capitalised.

File: docs/index.md [hunk 14]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
+Users interact with a registry by using docker push and pull commands. For example, `docker pull myregistry.com/stevvooe/batman:voice`.
 
-A registry is, at its heart, a collection of repositories. In turn, a repository
-is collection of images. Users interact with the registry by pushing images to
-or pulling images from the registry. The Docker Registry includes several
-optional features that you can configure according to your needs.
+Storage itself is delegated to drivers. The default storage driver is the local posix filesystem, which is suitable for development or small deployments. Additional cloud-based storage driver like S3 and Microsoft Azure are also supported. People looking into using other storage backends may do so by implementing their own driver implementing the [Storage API](storagedrivers.md).
 
-![](images/registry.png)
+Since securing access to your hosted images is paramount, the Registry natively supports TLS. You can also enforce basic authentication through a proxy like Nginx.
```
Comment (thaJeztah) [inline docs/index.md]: Not sure, but should Nginx be all uppercase? (NGINX) their site looks to use it that way http://nginx.com

File: docs/index.md [hunk 15]
```diff
@@ -4,54 +4,31 @@ page_description: Introduces the Docker Registry
...
 
-The architecture supports a configurable storage backend. You can store images
-on  a file system or on a service such as Amazon S3 or Microsoft Azure. The
-default storage system is the local disk; this is suitable for development or
-some small deployments.
+The Registry github repository includes reference implementations for additional authentication and authorization methods. Only very large or public deployments are expected to extend the Registry in this way.
```
Comment (thaJeztah) [inline docs/index.md]: GitHub

File: docs/index.md (hunk 8 above)
Comment (thaJeztah) [inline docs/index.md]: We still need a style guide on that, I think :)

File: docs/index.md [hunk 16]
```diff
@@ -59,10 +36,15 @@ communication.
...
+ - [Configuration](configuration.md)
+
+Advanced:
 
- - [Deploying a registry](deploying.md)
- - [Configure a registry](configuration.md)
  - [Storage driver model](storagedrivers.md)
  - [Working with notifications](notifications.md)
- - [Registry API v2](spec/api.md)
+ - [Registry API](spec/api.md)
```
Comment (thaJeztah) [inline docs/index.md]: Missing newline at end of file

Comment (dmp42) [inline docs/index.md]: Why would that be necessary?

File: docs/index.md (hunk 8 above)
Comment (dmp42) [inline docs/index.md]: I think this is a preoccupation of the 90s. :-)

File: docs/index.md (hunk 14 above)
Comment (dmp42) [inline docs/index.md]: nginx.org has title as "nginx" - wiki.nginx.org has title as "Nginx" - Wikipedia has it as "Nginx".

File: docs/index.md (hunk 13 above)
Comment (dmp42) [inline docs/index.md]: Agreed.

File: docs/index.md (hunk 15 above)
Comment (dmp42) [inline docs/index.md]: Ok.

**Comment by dmp42** [comment]
@thaJeztah nits are picked.

All: is this ready to merge?

File: docs/index.md (hunk 16 above)
Comment (thaJeztah) [inline docs/index.md]: In general, source-files should end with a newline. If a new line (text) is added at the end, it's only a single LOC, if the newline is missing, it will be two LOC's.

It's a nit but also convention (hence GitHub marking it with a red "missing newline" icon)

File: docs/index.md (hunk 8 above)
Comment (thaJeztah) [inline docs/index.md]: It *does* make it easier to add inline notes while reviewing on GitHub. Also, when reading the un-rendered Markdown (as plain-text), it's easier to read and better preserves the intent, irregardless of text-editor settings.

Maintaining the 80-chars wrap can be a PITA, so that's why we need to weigh the pros/cons of both and decide on a style-guide. Similar discussions (72-chars in that case) have been around formatting of commit-messages (http://stackoverflow.com/questions/2290016/git-commit-messages-50-72-formatting)

Comment (dmp42) [inline docs/index.md]: I wholeheartedly disagree with the benefits:

 * any decent text-editor will reflow *to your settings*: it's easier to read and better preserves the intent to *not* wrap at 80 chars, instead of breaking things for everyone else that doesn't have the same settings as you
 * the perceived benefit of making it easier to comment inline in github hides doc defects in the first place: if the text is too long, then the message is not clear enough, not well structured enough

Furthermore, enforcing 80-chars does introduce a lot of churn (by possibly triggering reformat of several unrelated lines), which is highly undesirable from a revision perspective.

To put it otherwise: this "rule" diverts resources and attention from what really matters (documentation content quality) toward pedantic and outdated churning.

Sorry for my feelings are so strong on this, but I really believe we should spend more energy fixing (a not-so-great) docker documentation overall, and less on devising infatuated guidelines :)

In the hope that does make sense and helps moving this debate forward.

Thanks for bringing this up @thaJeztah

File: docs/index.md (hunk 16 above)
Comment (dmp42) [inline docs/index.md]: That doesn't answer my question :-)

File: docs/index.md (hunk 8 above)
Comment (thaJeztah) [inline docs/index.md]: > Sorry for my feelings are so strong on this, but I really believe we should spend more energy fixing (a not-so-great) docker documentation overall, and less on devising infatuated guidelines :)

No need to apologize there; that's where debates/discussions are for. I fully agree that the docs are in need of a overhaul. The pace at which new features are added to docker (and projects 'integrated' to the docs) has resulted in a lot of duplicated/missing/incoherent texts, which makes it hard to read or find relevant information.

**Comment by moxiegirl** [comment]
There are issue here with capitalization.  `Docker Registry` is capitalized and `Registry` alone is not.  Seems like a nit I know but it is point in usability of the content...it allows a reader  to reader "to configure your registry" without pausing to think is this Registry or registry.  So, if by `Registry` you mean the `Docker Registry` those all need to be corrected here.

**Comment by moxiegirl** [comment]
@thaJeztah Thanks for the comments on the 80 characters. Having that  limit and enforcing that particularly with documentation is problematic.  Any company growing at our pace is going to have content issues, there are multiple reasons for that.  It seems like everyone is aware of that. We are working on mechanisms to make internal writers and external contributors more effective.  I'll bring up your comments with the OSS; please go ahead and make a proposal issue (if you don't already have one), that will raise visibility.

**Comment by dmp42** [comment]
@moxiegirl thanks.

So: the Registry (cap) is short for the Docker Registry (cap).
My Registry, your Registry, your Swarm - cap as well - correct?
An instance of the Registry - cap.

Do we agree on this?

Is there any case where we want to speak about "registry"? (and what would that be?)

Either way, this can be fixed later on. Merging this is probably more important.

**Comment by moxiegirl** [comment]
`Docker Registry` is a proper noun used to refer to our product.  When you want to emphasize its capabilities or make it the focus of a paragraph. Refer to it as the Docker Registry.  So, you might start the paragraph off as:

  *The Docker Registry has this feature...and so forth. The registry was developed.*

Readers understand by context that you are referring to the Docker Registry.  Similarly, if you are talking about a user's own registry implementation:

  *You can configure your registry. The registry recognizes its configuration.*

You shouldn't use Registry upper case at all. It confuses our branding and our readers.

**Comment by dmp42** [comment]
There is no such thing as a "user's own registry implementation".

Furthermore, "Docker Registry" is redundant. "Registry" is the product name - just like Docker Swarm is Swarm.

**Comment by moxiegirl** [comment]
Users configure and deploy a registry on their own networks.  I believe they would say "our registry" or "my registry" just as if they bought an Audi A8 and added a custom stereo --- they don't say "Audi's car" when referring to their A8.  

I checked upstream with the writers in the corporate communication area.  Apparently, we haven't yet protected our product names. Common words like Registry, Swarm, etc. are not typically subject to being trademarked/protected so people tend to put the company name in front of it to protect it.

**Comment by stevvooe** [comment]
LGTM!

If there are required fixes, please PR them separately.
